package/kedro_viz/models/flowchart.py

```diff
@@ -61,54 +69,69 @@ class GraphNodeType(str, Enum):
     )
 
 
-@dataclass(frozen=True)
-class ModularPipelineChild:
-    """Represent a child of a modular pipeline."""
+class ModularPipelineChild(BaseModel, frozen=True):
+    """Represent a child of a modular pipeline.
+
+    Args:
+        id (str): Id of the modular pipeline child
+        type (GraphNodeType): Type of modular pipeline child
+    """
 
     id: str
     type: GraphNodeType
 
 
-@dataclass
 class Tag(RegisteredPipeline):
```

Comment:
Maybe not related to this PR exactly but I don't understand why Tag is a RegisteredPipeline. Both are different from one another and maybe tag should have it's own id and name, instead of inheriting from RegisteredPipeline

Reply:
```
class Tag(BaseModel, frozen=True):
    """Represent a tag"""
    id: str
    name: str
```

Reply:
added a base class called NamedEntity (following the pattern in responses.py) for Tag/Registered Pipeline - https://github.com/kedro-org/kedro-viz/pull/1565/commits/6ce4e37e6d9cdd91cee7f2c4c17343acbcd29358